pkg/controller/operators/olm/operator.go

```diff
@@ -1211,6 +1239,99 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
 	return
 }
 
+func isNamespaceClusterScopedMap(operatorGroups ...*v1.OperatorGroup) map[string]bool {
+	result := map[string]bool{}
+	for _, operatorGroup := range operatorGroups {
+		result[operatorGroup.GetNamespace()] = NewNamespaceSet(operatorGroup.Status.Namespaces).IsAllNamespaces()
+	}
+	return result
+}
+
+func (a *Operator) syncOLMConfig(obj interface{}) (syncError error) {
+	a.logger.Info("Processing olmConfig")
+	olmConfig, ok := obj.(*v1.OLMConfig)
+	if !ok {
+		return fmt.Errorf("casting OLMConfig failed")
+	}
+
+	// Generate a map to track namespaces that are cluster scoped
+	operatorGroups, err := a.lister.OperatorsV1().OperatorGroupLister().List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	isNamespaceClusterScoped := isNamespaceClusterScopedMap(operatorGroups...)
+
+	// Get non-copied CSVs
+	requirement, err := labels.NewRequirement(v1alpha1.CopiedLabelKey, selection.DoesNotExist, []string{})
+	if err != nil {
+		return err
+	}
+
+	csvs, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().List(labels.NewSelector().Add(*requirement))
+	if err != nil {
+		return err
+	}
+
+	csvIsRequeued := false
+	for _, csv := range csvs {
+		// For each cluster scope installation
+		if !isNamespaceClusterScoped[csv.GetNamespace()] {
+			continue
+		}
+
+		// Get a count of the number of CSVS
+		requirement, err := labels.NewRequirement(v1alpha1.CopiedLabelKey, selection.Equals, []string{csv.GetNamespace()})
+		if err != nil {
+			return err
+		}
+
+		copiedCSVs, err := a.copiedCSVLister.List(labels.NewSelector().Add(*requirement))
+		if err != nil {
+			return err
+		}
+
+		// If the correct number of copied CSVs were found, continue
+		if len(copiedCSVs) == 0 == olmConfig.Spec.Features.DisableCopiedCSVs {
+			continue
+		}
+
+		// There were an incorrect number of copied CSVs, requeue the original CSV.
+		if err := a.csvQueueSet.Requeue(csv.GetNamespace(), csv.GetName()); err != nil {
```

Comment:
Would it be easier if this sync function directly deletes copied CSVs when the feature is disabled?

Reply:
spoke about this offline -- was convinced that letting the CSV controller handle this is more clean